tests/test_file_operations.py

```diff
@@ -0,0 +1,606 @@
+    assert not osp.exists(to_del)
+
+
+def test_remove_op_dir(test_dir):
+    """Test the operation to delete a directory"""
+
+    # Make a test file with dummy text
+    to_del = pathlib.Path(test_dir) / "dir_del"
+    os.mkdir(to_del)
+
+    parser = get_parser()
+    cmd = f"remove {to_del}"
+    args = cmd.split()
+    ns = parser.parse_args(args)
+
+    file_operations.remove(ns)
+
+    # Assert directory has been deleted
+    assert not osp.exists(to_del)
+
+
+def test_remove_op_bad_path(test_dir):
+    """Test that FileNotFoundError is raised when a bad path is given to the
+    operation to delete a file"""
+
+    to_del = pathlib.Path(test_dir) / "not_real.txt"
+
+    parser = get_parser()
+    cmd = f"remove {to_del}"
+    args = cmd.split()
+    ns = parser.parse_args(args)
+
+    with pytest.raises(FileNotFoundError) as ex:
+        file_operations.remove(ns)
+    assert "No such file or directory" in ex.value.args
+
+
+def test_remove_op_no_absolute():
+    """Test that ValueError is raised when a relative path
+    is given to the operation to delete a file"""
+
+    to_del = ".."
+
+    parser = get_parser()
+    cmd = f"remove {to_del}"
+    args = cmd.split()
+    ns = parser.parse_args(args)
+
+    with pytest.raises(ValueError) as ex:
+        file_operations.remove(ns)
+    assert f"path {to_del} must be absolute" in ex.value.args
+
+
+@pytest.mark.parametrize(
+    ["param_dict", "error_type"],
+    [
+        pytest.param(
+            {
+                "5": 10,
+                "FIRST": "SECOND",
+                "17": 20,
+                "65": "70",
+                "placeholder": "group leftupper region",
+                "1200": "120",
+            },
+            "None",
+            id="correct dict",
+        ),
+        pytest.param(
+            ["list", "of", "values"],
+            "TypeError",
+            id="incorrect dict",
+        ),
+        pytest.param({}, "ValueError", id="empty dict"),
+    ],
+)
+def test_configure_op(test_dir, fileutils, param_dict, error_type):
+    """Test configure operation with correct parameter dictionary, empty dicitonary, and an incorrect type"""
+
+    tag = ";"
+
+    conf_path = fileutils.get_test_conf_path(osp.join("tagged_tests", "marked/"))
+    # retrieve files to compare after test
+    correct_path = fileutils.get_test_conf_path(osp.join("tagged_tests", "correct/"))
+
+    # copy files to test directory
+    dir_util.copy_tree(conf_path, test_dir)
+    assert osp.isdir(test_dir)
+
+    tagged_files = sorted(glob(test_dir + "/*"))
+    correct_files = sorted(glob(correct_path + "*"))
+
+    # Pickle the dictionary
+    pickled_dict = pickle.dumps(param_dict)
+
+    # Encode the pickled dictionary with Base64
+    encoded_dict = base64.b64encode(pickled_dict)
+
+    # Run configure op on test files
+    for tagged_file in tagged_files:
+        parser = get_parser()
+        cmd = f"configure {tagged_file} {tagged_file} {tag} {encoded_dict}"
```

Comment:
yeah, in the case of this test, and previous tests, it would just overwrite the existing file

Reply:
Can we actually add a test where they are not the same path to avoid re-introducing [this](https://github.com/CrayLabs/SmartSim/pull/615#discussion_r1678471646) error?

Reply:
added